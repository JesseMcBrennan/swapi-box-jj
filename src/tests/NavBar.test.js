import React from 'react';
import { shallow } from 'enzyme';

import { NavBar } from '../Components/NavBar/index.js'

describe('NavBar', () => {
  let wrapper;
  let mockClick;
  let mockFavorites;
  let mockSelected;

  beforeEach(() => {
    mockClick = jest.fn()
    mockSelected = 'a button'
    mockFavorites = ['something']
    wrapper = shallow(
      <NavBar 
        handClick={mockClick}
        favoritesLength={mockFavorites}
        selectedButton={mockSelected}
      />)


  })

  it('renders the NavBar with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  })
})