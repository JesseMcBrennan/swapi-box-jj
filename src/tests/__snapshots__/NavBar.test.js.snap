// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders the NavBar with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    favoritesLength={3}
    handleClick={[Function]}
    selectedButton="a button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="unselected"
          name="people"
          onClick={[Function]}
>
          People
</button>,
        <button
          className="unselected"
          name="planets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          className="unselected"
          name="vehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          className="unselected"
          name="favorites"
          onClick={[Function]}
>
          Favorites 3
</button>,
      ],
      "className": "navBar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "unselected",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "unselected",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "unselected",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites 3",
          "className": "unselected",
          "name": "favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorites 3",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="unselected"
            name="people"
            onClick={[Function]}
>
            People
</button>,
          <button
            className="unselected"
            name="planets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            className="unselected"
            name="vehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            className="unselected"
            name="favorites"
            onClick={[Function]}
>
            Favorites 3
</button>,
        ],
        "className": "navBar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "unselected",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "unselected",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "unselected",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites 3",
            "className": "unselected",
            "name": "favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorites 3",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
